src/textual/app.py

```diff
@@ -156,6 +163,10 @@ def __init__(self) -> None:
         self._queue: Queue[str | None] = Queue(MAX_QUEUED_WRITES)
         self._file = sys.__stdout__
 
+        self._flushed = threading.Event()
```

Comment:
I suspect this is not the right approach.

At some point we need to tell the app to not make any more writes. With that in place, we can stop and join the thread. That way we can be certain there are no writes while we "flush".